xarray/core/common.py

```diff
@@ -810,11 +809,11 @@ def pipe(
             return func(self, *args, **kwargs)
 
     def rolling_exp(
-        self: T_DataWithCoords,
+        self: Self,
         window: Mapping[Any, int] | None = None,
         window_type: str = "span",
         **window_kwargs,
-    ) -> RollingExp[T_DataWithCoords]:
+    ) -> RollingExp[T_Xarray]:
```

Comment:
```suggestion
    ) -> RollingExp[Self]:
```

Reply:
Unfortunately this doesn't work, similar to the reason outlined below — `T_Xarray` isn't compatible with `T_DataWithCoords`...

Reply:
> Unfortunately this doesn't work, similar to the reason outlined below — `T_Xarray` isn't compatible with `T_DataWithCoords`...

I think thats why originally I was typing RollingExp using T_DataWithCoords